dbscan/structref.go

```diff
@@ -36,7 +36,10 @@ func getColumnToFieldIndexMap(structType reflect.Type) map[string][]int {
 				// Field is ignored, skip it.
 				continue
 			}
-			index := append(traversal.IndexPrefix, field.Index...)
+
+			index := make([]int, len(traversal.IndexPrefix), len(traversal.IndexPrefix)+len(field.Index))
+			copy(index, traversal.IndexPrefix)
+			index = append(index, field.Index...)
```

Comment:
Wouldn't it be cleaner and simpler like that?
```suggestion
			index := make([]int, 0, len(traversal.IndexPrefix)+len(field.Index))
			index = append(index, traversal.IndexPrefix...)
			index = append(index, field.Index...)
```

Reply:
sure! I like that version better, yes.